src/crimson/os/seastore/seastore_types.h

```diff
@@ -1298,6 +1650,77 @@ constexpr laddr_t L_ADDR_NULL = L_ADDR_MAX;
 constexpr laddr_t L_ADDR_ROOT = laddr_t::from_raw_uint(laddr_t::RAW_VALUE_MAX - 1);
 constexpr laddr_t L_ADDR_LBAT = laddr_t::from_raw_uint(laddr_t::RAW_VALUE_MAX - 2);
 
+struct laddr_printer_t {
+  const laddr_t &addr;
+};
+std::ostream &operator<<(std::ostream &, const laddr_printer_t &);
+
+enum class laddr_conflict_level_t {
+  object_content,  // only global metadata use this level
+  local_object_id,
+  local_clone_id,
+  block_offset,
+  never
+};
+
+enum class laddr_conflict_policy_t {
+  linear_search,
+  gen_random
```

Comment:
These names might be confusing by their own, how about adding some comments such as:

// see BtreeLBAManager::search_insert_pos(), linear will try the next iterator, and random will try gen_next_hint(hint)